src/wazuh_modules/vulnerability_detector/wm_vuln_detector.h

```diff
@@ -124,6 +124,36 @@
 // Types of scans
 typedef enum vu_scan_type_t { VU_BASELINE_SCAN , VU_FULL_SCAN, VU_PARTIAL_SCAN } vu_scan_type_t;
 
+// Type of tasks in vu_tasks_queue
+typedef enum vu_task_type_t {
+    VU_ON_DEMAND_SCAN,
+    VU_ON_DEMAND_FEED_UPDATE,
+    VU_INTERVAL_FEED_UPDATE,
+    VU_INTERVAL_SCAN,
+    VU_TASK_TYPES_SIZE // This should be the last constant
+} vu_task_type_t;
+
+// Status of tasks in vu_tasks_queue
+typedef enum vu_task_status_t {
+    VU_TASK_STATUS_VALID,
+    VU_TASK_STATUS_CANCELLED
+} vu_task_status_t;
+
+// Task context for vu_tasks_queue
+typedef struct vu_task_context_t {
+    size_t taskID;
+    vu_task_type_t task_type;
+    vu_task_status_t task_status;
+    char* feed_to_update;
+    int agent_to_scan;
+} vu_task_context_t;
+
+// Task flags for vu_tasks_queue
+extern bool VU_TASKS_QUEUE_FLAGS[VU_TASK_TYPES_SIZE];
```

Comment:
Is it necessary to declare this as `extern`?

Reply:
Fixed!